spring/src/main/java/io/seata/spring/annotation/GlobalTransactionScanner.java

```diff
@@ -304,28 +306,41 @@ public void setApplicationContext(ApplicationContext applicationContext) throws
         this.setBeanFactory(applicationContext);
     }
 
+    @Override
+    public Object postProcessBeforeInitialization(Object bean, String beanName) {
+        if (bean instanceof DataSourceProxy && ConfigurationFactory.getInstance().getBoolean(DATASOURCE_AUTOPROXY, false)) {
```

Comment:
I think you judge datasourceproxy targetDataSource is DataSourceProxy，If true, notice user.

Reply:
Deal with later